### PR TITLE
Implemented "Download a file to a temporary directory" example

### PR DESCRIPTION
**Pull request by budziq**
Implemented one of the examples "Download a file to a temporary directory" mentioned on https://github.com/brson/rust-cookbook/issues/24#issue-213203617

### CODE REVIEW

**Comment by dtolnay** [comment]
Nice work! I checked it off in the list in #24.

**Comment by kaboomshebang** [comment]
[The code example](https://rust-lang-nursery.github.io/rust-cookbook/web/clients/download.html#download-a-file-to-a-temporary-directory) does not compile (anymore). Not in the compiling playground, but also not locally. Completely new to Rust and not sure how to make this work :(

@budziq The errors pointed me to this [SO](https://stackoverflow.com/questions/53584631/fn-foo-result-throws-expected-2-type-arguments) thread, but I can't manage to get things working. Would it be possible for you to have a quick look at the example from the Rust Cookbook and maybe point me to a solution?

**Comment by AndyGauge** [comment]
Hi!  We recently updated the cookbook to edition 2018, that might be the issue, or it could be a couple other things.  
FIrst, try including in your Cargo.toml edition=2018 and let me know if that fixes it.  Also, can you include the playground link?  I really appreciate the feedback from readers of the book, and I want to help make this process easier.

**Comment by AndyGauge** [comment]
Oh! That is one of the recipes that we are in the process of updating to reqwest 0.10, and as such make sure your local Cargo.toml file references 0.9

The playground is locked to the latest version, so that work will really help here.

**Comment by AndyGauge** [comment]
Also, I think this is a working example, but it looks like the playground may not be friendly to actually making requests.  
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=f39a940aa008f83cc2c1eb33175306e7

**Comment by AndyGauge** [comment]
Alrighty, 10 hours later and the cookbook is updated!

https://rust-lang-nursery.github.io/rust-cookbook/web/clients/download.html

**Comment by kaboomshebang** [comment]
TLDR: I got the example working by specifying cargo.toml `tokio` like this:
`tokio = { version = "0.2.21", features = ["full"] }`
source: https://github.com/hyperium/hyper/issues/1891

@AndyGauge: First of all thanks for taking the time to help out! Much appreciated! :-)

https://rust-lang-nursery.github.io/rust-cookbook/web/clients/download.html
Two small more things:
- are all the dependencies for a code-example mentioned somewhere? Currently I have to add them by looking at the error log or at the use statements. Somehow including the `cargo.toml` would maybe help readers with running the examples.
- the crate-tag mentions using `tempdir` but actually it's using `tempfile`

Thanks again. Cheers

**Comment by AndyGauge** [comment]
The cookbook has a Cargo.toml file.  I don’t know if you can see it from the cookbook, but I’ve always looked here:https://github.com/rust-lang-nursery/rust-cookbook/blob/master/Cargo.toml That said, I think this is a big opportunity, we should at least include this sample Cargo.toml file in the book itself.  Can you create an issue for this? Sent from Mail for Windows 10 From: Kaboom ShebangSent: Monday, June 8, 2020 4:32 AMTo: rust-lang-nursery/rust-cookbookCc: Andrew Gauger; MentionSubject: Re: [rust-lang-nursery/rust-cookbook] Implemented "Download a file to a temporary directory" example (#79) TLDR: I got the example working by specifying cargo.toml tokio like this:tokio = { version = "0.2.21", features = ["full"] }source: hyperium/hyper#1891@AndyGauge: First of all thanks for taking the time to help out! Much appreciated! :-)https://rust-lang-nursery.github.io/rust-cookbook/web/clients/download.htmlTwo small more things:are all the dependencies for a code-example mentioned somewhere? Currently I have to add them by looking at the error log or at the use statements. Somehow including the cargo.toml would maybe help readers with running the examples.the crate-tag mentions using tempdir but actually it's using tempfileThanks again. Cheers—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by kaboomshebang** [comment]
Oh yes! That would be very helpful. Thanks. I will create an issue.
